### PR TITLE
fix: remove Issue resolution state (FPLA-2228)

### PR DESCRIPTION
**Pull request by Jamesnelson117**
**Before creating a pull request make sure that:**

- [x] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-2228

### Change description ###
Remove issue resolution state

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by akomisarek — APPROVED** [review]

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOService.java [hunk 1]
```diff
@@ -129,15 +130,11 @@ public State getStateBasedOnNextHearing(CaseData caseData, UUID cmoID) {
...
-                        return State.ISSUE_RESOLUTION;
-                    case FINAL:
-                        return State.FINAL_HEARING;
-                    default:
-                        return currentState;
-                }
+                && caseData.getReviewCMODecision().hasReviewOutcomeOf(SEND_TO_ALL_PARTIES)
+                && nextHearingBooking.get().isHearingOfType(HearingType.FINAL)) {
+                return State.FINAL_HEARING;
+            } else {
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOService.java:136]: else not needed here

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/HearingBooking.java [hunk 2]
```diff
@@ -80,4 +80,8 @@ public String toLabel(String dateFormat) {
         }
         return list;
     }
+
+    public boolean isHearingOfType(HearingType hearingType) {
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/HearingBooking.java:84]: minor: is it really worth having method for this? maybe better name would be isOfType(HearingType)

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOServiceTest.java [hunk 3]
```diff
@@ -261,7 +261,7 @@ void shouldThrowExceptionIfSealedCMOListIsEmpty() {
     }
 
     @Test
-    void shouldReturnIssueResolutionStateWhenNextHearingTypeIsIssueResolutionAndFeatureToggleIsToggledOn() {
+    void shouldReturnCaseManagementStateWhenNextHearingTypeIsIssueResolutionAndFeatureToggleIsToggledOn() {
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOServiceTest.java:264]: there is method shouldReturnCurrentStateWhenNextHearingIsNotOfTypeIssueResolutionOrFinal which should be renamed

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/HearingBooking.java (hunk 2 above)
Comment (akomisarek) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/HearingBooking.java:84]: Reads nicely :)

**Review by akomisarek — APPROVED** [review]

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOServiceTest.java (hunk 3 above)
Comment (Jamesnelson117) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOServiceTest.java:264]: Few of them actually, i'll change.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOServiceTest.java [hunk 4]
```diff
@@ -293,7 +293,7 @@ void shouldReturnFinalHearingStateWhenNextHearingTypeIsFinalAndFeatureToggleIsTo
     }
 
     @Test
-    void shouldReturnCurrentCaseStateWhenNextHearingIsTypeIssueResolutionAndFeatureToggleIsToggledOff() {
+    void shouldReturnCaseManagementStateWhenNextHearingIsTypeIssueResolutionAndFeatureToggleIsToggledOff() {
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOServiceTest.java:296]: is**Of**Type

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOServiceTest.java [hunk 5]
```diff
@@ -304,7 +304,7 @@ void shouldReturnCurrentCaseStateWhenNextHearingIsTypeIssueResolutionAndFeatureT
     }
 
     @Test
-    void shouldReturnCurrentStateWhenNextHearingIsNotOfTypeIssueResolutionOrFinal() {
+    void shouldReturnCaseManagementStateWhenNextHearingIsNotOfTypeIssueResolutionOrFinal() {
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOServiceTest.java:307]: IsNotOfFinalType

**Review by tomaszpowroznik — APPROVED** [review]

**Review by tomaszpowroznik — APPROVED** [review]

**Pull request merged**